src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -355,34 +430,56 @@ public static FormulaValue Mid(IRContext irContext, FormulaValue[] args)
 
         public static FormulaValue Left(IRContext irContext, FormulaValue[] args)
         {
-            var source = (StringValue)args[0];
-            var count = (NumberValue)args[1];
+            return LeftOrRight(irContext, args, Left);
+        }
+
+        public static FormulaValue Right(IRContext irContext, FormulaValue[] args)
+        {
+            return LeftOrRight(irContext, args, Right);
+        }
 
-            if (count.Value >= source.Value.Length)
+        private static string Left(string str, int i)
+        {
+            if (i >= str.Length)
             {
-                return source;
+                return str;
             }
 
-            return new StringValue(irContext, source.Value.Substring(0, (int)count.Value));
+            return str.Substring(0, i);
         }
 
-        public static FormulaValue Right(IRContext irContext, FormulaValue[] args)
+        private static string Right(string str, int i)
         {
-            var source = (StringValue)args[0];
-            var count = (NumberValue)args[1];
+            if (i >= str.Length)
+            {
+                return str;
+            }
+
+            return str.Substring(str.Length - i);
+        }
 
-            if (count.Value == 0)
+        private static FormulaValue LeftOrRight(IRContext irContext, FormulaValue[] args, Func<string, int, string> leftOrRight)
+        {
+            if (args[0] is BlankValue || args[1] is BlankValue)
             {
                 return new StringValue(irContext, string.Empty);
             }
 
-            if (count.Value >= source.Value.Length)
+            if (args[1] is not NumberValue)
```

Comment:
>args[1] is not NumberValue [](http://example.com/codeflow?start=16&length=26)

`if (args[1] is not NumberValue count)` - and you won't need the declaration/cast on line 474

Reply:
Good catch!